source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/wrappers/torchrl/torchrl_env_wrapper.py

```diff
@@ -0,0 +1,601 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import contextlib
+import numpy as np
+import os
+import time
+import torch
+import warnings
+from collections.abc import Iterator
+from dataclasses import asdict
+from tensordict import TensorDict, TensorDictBase, unravel_key
+from tensordict.nn import ProbabilisticTensorDictSequential, TensorDictModule, dispatch
+from typing import Any
+
+from torchrl.collectors import SyncDataCollector
+from torchrl.collectors.utils import split_trajectories
+from torchrl.data.tensor_specs import CompositeSpec, UnboundedContinuousTensorSpec
+from torchrl.envs import GymWrapper
+from torchrl.envs.libs.gym import default_info_dict_reader
+from torchrl.envs.utils import _terminated_or_truncated
+from torchrl.objectives import ClipPPOLoss
+from torchrl.objectives.utils import _reduce
+from torchrl.record.loggers.wandb import WandbLogger
+from torchrl.trainers import Trainer
+from torchrl.trainers.trainers import LOGGER_METHODS
+
+from omni.isaac.lab.envs import ManagerBasedRLEnv
+
+
+class TorchRLEnvWrapper(GymWrapper):
```

Comment:
I think we can get around most of the extra features here with torchrl transforms and by telling GymWrapper that the env is auto-resetting - happy to help getting this done. 
That will be more robust long term, as this implementation relies on a bunch of private features of torchrl that we could modify in the future.